sonic_platform_base/sonic_xcvr/api/public/sff8436.py

```diff
@@ -0,0 +1,252 @@
+            "txbiaslowalarm": float("{:.3f}".format(tx_bias_thresholds[consts.TX_BIAS_LOW_ALARM_FIELD])),
+            "txbiashighwarning": float("{:.3f}".format(tx_bias_thresholds[consts.TX_BIAS_HIGH_WARNING_FIELD])),
+            "txbiaslowwarning": float("{:.3f}".format(tx_bias_thresholds[consts.TX_BIAS_LOW_WARNING_FIELD]))
+        }
+
+    def get_rx_los(self):
+        rx_los = self.xcvr_eeprom.read(consts.RX_LOS_FIELD)
+        if rx_los is None:
+            return None
+        return [bool(rx_los & (1 << i)) for i in range(self.NUM_CHANNELS)]
+
+    def get_tx_fault(self):
+        tx_fault = self.xcvr_eeprom.read(consts.TX_FAULT_FIELD)
+        if tx_fault is None:
+            return None
+        return [bool(tx_fault & (1 << i)) for i in range(self.NUM_CHANNELS)]
+
+    def get_tx_disable(self):
+        tx_disable = self.xcvr_eeprom.read(consts.TX_DISABLE_FIELD)
+        if tx_disable is None:
+            return None
+        return [bool(tx_disable & (1 << i)) for i in range(self.NUM_CHANNELS)]
+
+    def get_tx_disable_channel(self):
+        return self.xcvr_eeprom.read(consts.TX_DISABLE_FIELD)
+
+    def get_temperature(self):
+        temp = self.xcvr_eeprom.read(consts.TEMPERATURE_FIELD)
+        if temp is None:
+            return None
+        return float("{:.3f}".format(temp))
+
+    def get_voltage(self):
+        voltage = self.xcvr_eeprom.read(consts.VOLTAGE_FIELD)
+        if voltage is None:
+            return None
+        return float("{:.3f}".format(voltage))
+
+    def get_tx_bias(self):
+        tx_bias = self.xcvr_eeprom.read(consts.TX_BIAS_FIELD)
+        if tx_bias is None:
+            return None
+        return [channel_bias for channel_bias in tx_bias.values()]
+
+    def get_rx_power(self):
+        rx_power = self.xcvr_eeprom.read(consts.RX_POWER_FIELD)
+        if rx_power is None:
+            return None
+        return [float("{:.3f}".format(channel_power)) for channel_power in rx_power.values()]
+
+    def get_tx_power(self):
+        return ["N/A" for _ in range(self.NUM_CHANNELS)]
+
+    def tx_disable(self, tx_disable):
+        val = 0xF if tx_disable else 0x0
+        return self.xcvr_eeprom.write(consts.TX_DISABLE_FIELD, val)
+
+    def tx_disable_channel(self, channel, disable):
+        channel_state = self.get_tx_disable_channel()
+        if channel_state is None:
+            return False
+
+        for i in range(self.NUM_CHANNELS):
+            mask = (1 << i)
+            if not (channel & mask):
+                continue
+            if disable:
+                channel_state |= mask
+            else:
+                channel_state &= ~mask
+
+        return self.xcvr_eeprom.write(consts.TX_DISABLE_FIELD, channel_state)
+
+    def get_power_override(self):
+        return self.xcvr_eeprom.read(consts.POWER_OVERRIDE_FIELD)
+
+    def set_power_override(self, power_override, power_set):
+        return self.xcvr_eeprom.write(consts.POWER_OVERRIDE_FIELD, power_override) and \
```

Comment:
power_set is effective only when power_overrride is '1'. you can save one write by checking:

```
write(consts.POWER_OVERRIDE_FIELD, power_override)
if power_override:  
          write(consts.POWER_SET_FIELD, power_set) 
```

Reply:
Done